planning/mission_planner/src/rerouting_static_obstacle/rerouting_static_obstacle.cpp

```diff
@@ -0,0 +1,203 @@
+    std::bind(&ReroutingStaticObstacle::map_callback, this, std::placeholders::_1));
+
+  sub_trigger_ = this->create_subscription<geometry_msgs::msg::PointStamped>(
+    "input/reroute_point", rclcpp::QoS(1),
+    std::bind(&ReroutingStaticObstacle::on_trigger, this, _1));
+
+  const auto adaptor = component_interface_utils::NodeAdaptor(this);
+  adaptor.init_cli(cli_change_route_);
+}
+
+void ReroutingStaticObstacle::on_odom(const nav_msgs::msg::Odometry::SharedPtr msg)
+{
+  current_pose_ = msg->pose.pose;
+}
+
+void ReroutingStaticObstacle::route_callback(
+  const autoware_planning_msgs::msg::LaneletRoute::ConstSharedPtr msg)
+{
+  goal_pose_ = msg->goal_pose;
+}
+
+void ReroutingStaticObstacle::map_callback(const HADMapBin::ConstSharedPtr msg)
+{
+  route_handler_.setMap(*msg);
+  lanelet_map_ptr_ = std::make_shared<lanelet::LaneletMap>();
+  lanelet::utils::conversion::fromBinMsg(
+    *msg, lanelet_map_ptr_, &traffic_rules_ptr_, &routing_graph_ptr_);
+  lanelet::ConstLanelets all_lanelets = lanelet::utils::query::laneletLayer(lanelet_map_ptr_);
+  road_lanelets_ = lanelet::utils::query::roadLanelets(all_lanelets);
+}
+
+void ReroutingStaticObstacle::on_trigger(const geometry_msgs::msg::PointStamped::SharedPtr msg)
+{
+  geometry_msgs::msg::Point selected_point{msg->point};
+
+  lanelet::ConstLanelets selected_point_lanelets;
+
+  auto selected_point_lanelet_found =
+    get_selected_point_lanelets(selected_point, selected_point_lanelets);
+
+  lanelet::ConstLanelets remaining_route_lanelets;
+  auto remaining_route_lanelets_found = get_remaining_route_lanelets(remaining_route_lanelets);
+
+  lanelet::ConstLanelet selected_point_lanelet;
+  bool selected_point_in_route{false};
+  if (selected_point_lanelet_found && remaining_route_lanelets_found) {
+    selected_point_in_route = is_selected_point_in_route(
+      selected_point_lanelets, remaining_route_lanelets, selected_point_lanelet);
+  } else {
+    return;
+  }
+
+  if (selected_point_in_route) {
+    lanelet::routing::LaneletPath alternative_route_lanelets;
+    auto alternative_route_found =
+      search_alternative_route(selected_point_lanelet, alternative_route_lanelets);
+
+    if (alternative_route_found) {
+      change_route(alternative_route_lanelets);
+    }
+  }
+}
+
+bool ReroutingStaticObstacle::get_selected_point_lanelets(
+  const geometry_msgs::msg::Point & selected_point,
+  lanelet::ConstLanelets & selected_point_lanelets) const
+{
+  if (!lanelet::utils::query::getCurrentLanelets(
+        road_lanelets_, selected_point, &selected_point_lanelets)) {
+    RCLCPP_WARN_STREAM(this->get_logger(), "Failed to find selected_point current lanelet.");
+    return false;
+  }
+
+  return true;
+}
+
+bool ReroutingStaticObstacle::get_remaining_route_lanelets(
+  lanelet::ConstLanelets & remaining_route_lanelets) const
+{
+  return route_handler_.planPathLaneletsBetweenCheckpoints(
+    current_pose_, goal_pose_, &remaining_route_lanelets);
+}
```

Comment:
Does this require calculating the route again? What about cropping from current_lanes onwards from the current lanes?